From b0080bd8a32d77a59b4c970d0107f98ec7a159ed Mon Sep 17 00:00:00 2001
From: dseefeld <dseefeld@microsoft.com>
Date: Thu, 11 Jun 2020 12:15:06 -0400
Subject: [PATCH] Include netcorapp5.0 versions in libraries nupkgs

Assets for netcoreapp3.0 are not built in source-build because we don't
have references for past netcoreapp versions.  That leaves the packages
built by the libraries build with a placeholder for netcoreapp3.0, which
causes issues with consumption of these packages in dependent repos.
Packages for netcoreapp5.0 are built but not packaged by design.  This
patch includes these assets in the packages built by the libraries
build, only for the source-build build.
---
 src/libraries/Directory.Build.targets | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libraries/Directory.Build.targets b/src/libraries/Directory.Build.targets
index 0548b4468af..7460e44616b 100644
--- a/src/libraries/Directory.Build.targets
+++ b/src/libraries/Directory.Build.targets
@@ -230,7 +230,7 @@
   </Choose>
 
   <PropertyGroup>
-    <ExcludeFromPackage Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', '$(NetCoreAppCurrent)')) and '$(ExcludeCurrentNetCoreAppFromPackage)' == 'true'">true</ExcludeFromPackage>
+    <ExcludeFromPackage Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', '$(NetCoreAppCurrent)')) and '$(ExcludeCurrentNetCoreAppFromPackage)' == 'true' and '$(DotNetBuildFromSource)' != 'true'">true</ExcludeFromPackage>
   </PropertyGroup>
 
   <!-- Adds System.Runtime.Versioning*Platform* annotation attributes to < .NET 5 builds -->
-- 
2.25.4

